src/lib.rs

```diff
@@ -388,8 +388,50 @@ impl KmerCountTable {
         Ok(())
     }
 
-    pub fn kmers_and_hashes(&self, seq: String, allow_bad_kmers: bool) -> PyResult<Vec<(String, u64)>> {
-        Ok(vec![])
+    pub fn kmers_and_hashes(
+        &self,
+        seq: String,
+        allow_bad_kmers: bool,
+    ) -> PyResult<Vec<(String, u64)>> {
+        let seqb = seq.as_bytes();
+        let mut hasher = SeqToHashes::new(
+            seqb,
+            self.ksize.into(),
+            allow_bad_kmers,
+            false,
+            HashFunctions::Murmur64Dna,
+            42,
+        );
+
+        let ksize = self.ksize as usize;
+        let end: usize = seq.len() - ksize + 1;
+
+        let mut v: Vec<(String, u64)> = vec![];
+        for start in 0..end {
+            let substr = &seq[start..start + ksize];
+            let hashval = hasher.next().unwrap().unwrap();
+            v.push((substr.to_string(), hashval));
```

Comment:
Check what happens here if `allow_bad_kmers` is False and a bad kmer is encountered?

Is hashval null? error?

Reply:
Can hashes get out of sync with substr slices if a bad kmer is skipped?

Reply:
this is where the rust error handling is nice - unwrap() will just panic! and then there are ways of handling the rest when we get to it.

I'll write various tests to make sure we cover the cases.